app/models/spree/image_decorator.rb

```diff
@@ -0,0 +1,54 @@
+Spree::Image.class_eval do
+  has_attached_file :attachment,
```

Comment:
This looks like a significant change, since `Spree::Image` is used for images on product and collection pages as well.

Reply:
we need to `load/unload` image styles by demand, what do you think?

Reply:
Thanks for that detail, after re-reading the decorator I think I see what you are trying to accomplish. Are you calling `reprocess_images!` at any point? I think that will be required if one of those flags goes from false to true.

Reply:
I was close: https://github.com/thoughtbot/paperclip/wiki/Thumbnail-Generation#generatingregenerating-your-thumbnails

Reply:
yes, here https://github.com/geminimvp/engine_storefront/pull/455/files#diff-5c09dc27581ae9055764d70ef1e004a0R13